### DISCUSSION TITLE
tutorials/scripting/change_scenes_manually

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#90] [User-contributed notes]
# tutorials/scripting/change_scenes_manually

Sometimes it helps to have more control over how you swap scenes around. A Viewport's child nodes will render to the image it generates. This holds true even for nodes outside of the "current" scen...

https://docs.godotengine.org/en/stable/tutorials/scripting/change_scenes_manually.html

**Comment by leonardoraele** [comment]
When changing scenes manually, it might be a good idea to set `get_tree().current_scene = scene` so that scripts that rely on `SceneTree.current_scene` don't break.

**Comment by RotterAxel** [comment]
This doc needs to have an added GetTree().ChangeSceneToFile("res://world.tscn"); 
I think it does the same thing as loading a new scene and deleting the old one.

**Comment by sjrct** [comment]
I ended up writing this function to change the scene to a particular node (so I can set it up a little before switching to it). You have to assign the tree to a variable as `get_tree()` will return null once `remove_child` is called (after all, the current node should be at that time removed from the tree).

```gdscript
func change_scene_to_node(node):
	var tree = get_tree()
	var cur_scene = tree.get_current_scene()
	tree.get_root().add_child(node)
	tree.get_root().remove_child(cur_scene)
	tree.set_current_scene(node)
```

**Reply by FireCatMagic** [reply]
Alternatively, Engine.get_main_loop() works to access the SceneTree from anywhere in cide
